tools/automatic_query_fixer/src/test/java/com/google/cloud/bigquery/utils/queryfixer/BigQueryServiceTest.java

```diff
@@ -69,4 +69,70 @@ public void listTables() {
     assertEquals(2, tables.size());
     assertThat(tables, contains("311_request", "311_service_requests"));
   }
+
+  /**
+   * Mock the connection to the BigQuery server. Use reflection to inject the fake connection into
+   * the {@link BigQueryService}.
+   */
+  private void fakeBigQuery() {
+    BigQuery bigQuery = mock(BigQuery.class);
+    correctShakespeare(bigQuery);
+    incorrectShakespeare(bigQuery);
+    fakeTables(bigQuery);
+    try {
+      FieldUtils.writeField(service, "bigQuery", bigQuery, true);
+    } catch (IllegalAccessException ignored) {
+    }
+  }
+
+  /**
+   * Mock the response of dry run if "SELECT corpus FROM `bigquery-public-data.samples.shakespeare`
+   * GROUP BY corpus limit 1000" is given.
+   */
+  private void correctShakespeare(BigQuery bigQuery) {
+    String query =
+        "SELECT corpus FROM `bigquery-public-data.samples.shakespeare` GROUP BY corpus limit 1000";
+    QueryJobConfiguration queryConfig =
+        QueryJobConfiguration.newBuilder(query).setDryRun(true).build();
+
+    Job job = mock(Job.class);
```

Comment:
Why you need to mock this simple object? you can create a default instance job instance